src/components/shared/ShowHideFunctionality.tsx

```diff
@@ -0,0 +1,71 @@
+import { IconButton, TextField } from '@mui/material';
+import InputAdornment from '@mui/material/InputAdornment';
+
+import React from 'react';
+import {  FieldErrors } from 'react-hook-form';
+
+interface ShowHideFunctionalityProps {
+    field:  any;
+    label: string;
+    id: string;
+    name: string;
+    required: boolean;
+    VisibilityOff: React.ElementType;
+    Visibility: React.ElementType;
+    color?: string
+    errors: FieldErrors<any>
+}
+
+
+export default function ShowHideFunctionality(
+    {
+        field,
+        label,
+        id,
+        name,
+        required,
+        VisibilityOff,
+        Visibility,
+        color,
+        errors
+    }: ShowHideFunctionalityProps
+
+) {
+    const [showPassword, setShowPassword] = React.useState(false);
+
+    const handleClickShowPassword = () => setShowPassword((show) => !show)
+
+    const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.preventDefault()
+    }
```

Comment:
make these more generic and not `password` specific.

Reply:
Fixed